website/public/js/controllers/tasksCtrl.js

```diff
@@ -76,10 +76,23 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
      * Pushes task to top or bottom of list
      */
     $scope.pushTask = function(task, index, location) {
-      var to = (location === 'bottom') ? -1 : 0;
+      var to = (location === 'bottom' || $scope.pushLocation === 'bottom') ? -1 : 0;
       User.user.ops.sortTask({params:{id:task.id},query:{from:index, to:to}})
     };
 
+    var ctrlKeys = [17, 224, 91];
+    $scope.$on("habit:keydown", function (e, keyEvent) {
+      if (ctrlKeys.indexOf(keyEvent.keyCode) !== -1) {
+        $scope.pushLocation = "bottom";
+      }
+    });
+
+    $scope.$on("habit:keyup", function (e, keyEvent) {
+      if (ctrlKeys.indexOf(keyEvent.keyCode) !== -1) {
+        $scope.pushLocation = "top";
+      }
+    });
+
```

Comment:
Can you change the words "pushLocation" and "top" and "bottom" to something more generic? If I understand your code correctly, then `$scope.pushLocation` is set based on whether the ctrl/cmd key is pressed or not. We're using that here for pushing tasks, but it could be used elsewhere in future for other purposes. A generic name like 'ctrlPressed' with values of 'true' or 'false' would make more sense in other contexts. Lines 83 t0 95 could then be moved to a more generic location in the file, since they aren't necessarily specific to pushing tasks.